src/crypto/NoteEncryption.cpp

```diff
@@ -41,7 +41,7 @@ void KDF(unsigned char *K,
 }
 
 template<size_t MLEN>
-NoteEncryption<MLEN>::NoteEncryption(uint256 seed) : times_ran(0), seed(seed) {
+NoteEncryption<MLEN>::NoteEncryption(uint256 seed) : nonce(0), seed(seed) {
```

Comment:
I don't think it hurts anything for it to be stateful and provide reviewers with some guarantee about downstream codepaths.

In fact, the decryptor should even eventually be stateful so that you only need to supply `sk_enc` once; right now it will generate `pk_enc` from your `sk_enc` every time you attempt to decrypt a ciphertext.